From 35678275294b2b70c23d61c59e2f75d8e9a992c1 Mon Sep 17 00:00:00 2001
From: yeyiyang <yeyiyang3@h-partners.com>
Date: Mon, 17 Nov 2025 17:22:24 +0800
Subject: [PATCH 804/960] downstream: arm64: add config for cpuidle-haltpoll

Add config for cpuidle-haltpoll. This feature is disabled by
default. Enable it using the following command:
echo Y > /sys/module/cpuidle_haltpoll/parameters/force

Signed-off-by: yeyiyang <yeyiyang3@h-partners.com>
Signed-off-by: Liu Shixin <liushixin2@huawei.com>
---
 arch/arm64/Kconfig | 2 ++
 config.aarch64     | 3 +++
 config.aarch64-64k | 3 +++
 config.x86_64      | 2 +-
 4 files changed, 9 insertions(+), 1 deletion(-)

diff --git a/arch/arm64/Kconfig b/arch/arm64/Kconfig
index 8a91ba0f21e9..289caf41b926 100644
--- a/arch/arm64/Kconfig
+++ b/arch/arm64/Kconfig
@@ -36,6 +36,8 @@ config ARM64
 	select ARCH_HAS_NON_OVERLAPPING_ADDRESS_SPACE
 	select ARCH_HAS_NONLEAF_PMD_YOUNG if ARM64_HAFT
 	select ARCH_HAS_OPTIMIZED_POLL
+	select ARCH_CPUIDLE_HALTPOLL
+	select HALTPOLL_CPUIDLE
 	select ARCH_HAS_PTE_DEVMAP
 	select ARCH_HAS_PTE_SPECIAL
 	select ARCH_HAS_SETUP_DMA_OPS
diff --git a/config.aarch64 b/config.aarch64
index a882970db60b..6e3637a865ba 100644
--- a/config.aarch64
+++ b/config.aarch64
@@ -573,6 +573,7 @@ CONFIG_ENERGY_MODEL=y
 CONFIG_ARCH_HIBERNATION_POSSIBLE=y
 CONFIG_ARCH_HIBERNATION_HEADER=y
 CONFIG_ARCH_SUSPEND_POSSIBLE=y
+CONFIG_ARCH_CPUIDLE_HALTPOLL=y
 # end of Power management options
 
 #
@@ -592,6 +593,7 @@ CONFIG_CPU_IDLE_GOV_MENU=y
 #
 # CONFIG_ARM_PSCI_CPUIDLE is not set
 # end of ARM CPU Idle Drivers
+CONFIG_HALTPOLL_CPUIDLE=y
 # end of CPU Idle
 
 #
@@ -729,6 +731,7 @@ CONFIG_TRACE_IRQFLAGS_SUPPORT=y
 CONFIG_TRACE_IRQFLAGS_NMI_SUPPORT=y
 CONFIG_HAVE_ARCH_TRACEHOOK=y
 CONFIG_HAVE_DMA_CONTIGUOUS=y
+CONFIG_ARCH_HAS_OPTIMIZED_POLL=y
 CONFIG_GENERIC_SMP_IDLE_THREAD=y
 CONFIG_GENERIC_IDLE_POLL_SETUP=y
 CONFIG_ARCH_HAS_FORTIFY_SOURCE=y
diff --git a/config.aarch64-64k b/config.aarch64-64k
index 4706984f2437..55c702a876c0 100644
--- a/config.aarch64-64k
+++ b/config.aarch64-64k
@@ -575,6 +575,7 @@ CONFIG_ENERGY_MODEL=y
 CONFIG_ARCH_HIBERNATION_POSSIBLE=y
 CONFIG_ARCH_HIBERNATION_HEADER=y
 CONFIG_ARCH_SUSPEND_POSSIBLE=y
+CONFIG_ARCH_CPUIDLE_HALTPOLL=y
 # end of Power management options
 
 #
@@ -594,6 +595,7 @@ CONFIG_CPU_IDLE_GOV_MENU=y
 #
 # CONFIG_ARM_PSCI_CPUIDLE is not set
 # end of ARM CPU Idle Drivers
+CONFIG_HALTPOLL_CPUIDLE=y
 # end of CPU Idle
 
 #
@@ -731,6 +733,7 @@ CONFIG_TRACE_IRQFLAGS_SUPPORT=y
 CONFIG_TRACE_IRQFLAGS_NMI_SUPPORT=y
 CONFIG_HAVE_ARCH_TRACEHOOK=y
 CONFIG_HAVE_DMA_CONTIGUOUS=y
+CONFIG_ARCH_HAS_OPTIMIZED_POLL=y
 CONFIG_GENERIC_SMP_IDLE_THREAD=y
 CONFIG_GENERIC_IDLE_POLL_SETUP=y
 CONFIG_ARCH_HAS_FORTIFY_SOURCE=y
diff --git a/config.x86_64 b/config.x86_64
index 9086845db0c7..56fed63d3e2c 100644
--- a/config.x86_64
+++ b/config.x86_64
@@ -306,7 +306,7 @@ CONFIG_GENERIC_BUG=y
 CONFIG_GENERIC_BUG_RELATIVE_POINTERS=y
 CONFIG_ARCH_MAY_HAVE_PC_FDC=y
 CONFIG_GENERIC_CALIBRATE_DELAY=y
-CONFIG_ARCH_HAS_CPU_RELAX=y
+CONFIG_ARCH_HAS_OPTIMIZED_POLL=y
 CONFIG_ARCH_HIBERNATION_POSSIBLE=y
 CONFIG_ARCH_SUSPEND_POSSIBLE=y
 CONFIG_AUDIT_ARCH=y
-- 
2.43.0

